tools/loadtest/metrics/compare-metrics.sh

```
@@ -0,0 +1,559 @@
+# Collect all candidate files, sorted by collected_at timestamp (newest first)
+gather_files() {
+  local dir="$1"
+  if [[ ! -d "$dir" ]]; then
+    echo "Error: Metrics directory '$dir' does not exist." >&2
+    exit 1
+  fi
+
+  # Find all .json files, extract collected_at, sort descending by timestamp
+  local tmpfile
+  tmpfile=$(mktemp)
+  trap "rm -f '$tmpfile'" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)